### PR TITLE
cli: warning prompt yes/no before volume deletion

### PR DESCRIPTION
**Pull request by oshankkumar**
closes: #1163

Signed-off-by: Oshank Kumar <okumar@redhat.com>

### CODE REVIEW

**Comment by oshankkumar** [comment]
retest this please

**Comment by Madhu-1** [comment]
@oshankkumar I think we should not provide this flag, if we provide warning prompt here, we should provide for all delete operations.
@aravindavk @prashanthpai @atinmu whats your thought on this?

**Comment by aravindavk** [comment]
> I think we should not provide this flag, if we provide warning prompt here, we should provide for all delete operations as well.

Yes, required for all destructive operations. But it can be incremental, other commands can be added in different PRs.

Also in `gluster`, these prompts can be disabled using `--mode=script`(or something similar).

File: glustercli/cmd/volume.go [hunk 1]
```diff
@@ -218,6 +219,9 @@ var volumeDeleteCmd = &cobra.Command{
 	Args:  cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
 		volname := cmd.Flags().Args()[0]
+		if ok := prompt.Confirm("Are you sure you want to delete volume %s [yes/no]", volname); !ok {
```
Comment (Madhu-1) [inline glustercli/cmd/volume.go]: question: why can't we implement this functionality in glusterd2 instead of importing external package for this?

File: Gopkg.lock [hunk 2]
```diff
@@ -354,6 +360,12 @@
   packages = ["xdr2"]
   revision = "1a41d1a06c93bf8a0e0385be3847a277bb793187"
 
+[[projects]]
+  branch = "master"
```
Comment (Madhu-1) [inline Gopkg.lock]: we should not manually edit `Gopkg.lock` file, need to add this dependency if required in `Gopkg.toml` file and regenerate this file.

File: glustercli/cmd/volume.go (hunk 1 above)
Comment (oshankkumar) [reply glustercli/cmd/volume.go]: this doesn't require much functionality, we can easily implement this.

File: Gopkg.lock (hunk 2 above)
Comment (oshankkumar) [reply Gopkg.lock]: when I regenerate this file, it default revision which it took was not working. so I modified the revision with the current one.

Comment (Madhu-1) [reply Gopkg.lock]: but we should not edit lock file, anyhow you are implementing the function for displaying prompt we don't require this one.

File: glustercli/cmd/utils.go [hunk 3]
```diff
@@ -54,3 +55,22 @@ func sizeToMb(value string) (uint64, error) {
...
+	var s string
+	fmt.Printf(prompt+"? ", args...)
+	fmt.Scanln(&s)
+	return s
+}
+
+// PromptConfirm prompts for confirmation
+func PromptConfirm(prompt string, args ...interface{}) bool {
+	for {
+		switch readString(prompt, args...) {
```
Comment (Madhu-1) [inline glustercli/cmd/utils.go]: convert the `args` string to lower case of before switch statement

File: glustercli/cmd/root.go [hunk 4]
```diff
@@ -72,6 +73,7 @@ type GlustercliOption struct {
 //AddPersistentFlag will initialize the Global Flags of root command.
 func (gOpt *GlustercliOption) AddPersistentFlag(flagSet *pflag.FlagSet) {
 	// Global flags, applicable for all sub commands
+	flagSet.BoolVarP(&gOpt.ScriptMode, "script-mode", "", false, "running in script mode")
```
Comment (Madhu-1) [inline glustercli/cmd/root.go]: as we are not using `shorthand` change `BoolVarP` to `BoolVar`

File: Gopkg.lock [hunk 5]
```diff
@@ -580,6 +580,6 @@
 [solve-meta]
   analyzer-name = "dep"
   analyzer-version = 1
-  inputs-digest = "6c5bb30b3fc2f361b83a245f51cd824ba49f4e46a125ca38702c8b561cd5dd83"
+  inputs-digest = "62a12e75f5497a585e2cf71c56e6129dbdec6d69886bb10e4346eb67008de949"
```
Comment (Madhu-1) [inline Gopkg.lock]: why is this change required?

File: glustercli/cmd/utils.go (hunk 3 above)
Comment (aravindavk) [inline glustercli/cmd/utils.go]: use `bufio.ReadString('\n')`

Comment (aravindavk) [reply glustercli/cmd/utils.go]: reader := bufio.NewReader(os.Stdin)
    resp, _ := reader.ReadString('\n')                                                                                                                                                        
    switch strings.TrimSpace(strings.ToLower(resp)) {                                                                                                                                         
    case "y", "yes":                                                                                                                                                                          
        return true                                                                                                                                                                           
    default:                                                                                                                                                                                  
        return false                                                                                                                                                                          
    }

File: glustercli/cmd/utils.go [hunk 6]
```diff
@@ -54,3 +55,22 @@ func sizeToMb(value string) (uint64, error) {
...
+func readString(prompt string, args ...interface{}) string {
+	var s string
+	fmt.Printf(prompt+"? ", args...)
+	fmt.Scanln(&s)
+	return s
+}
+
+// PromptConfirm prompts for confirmation
+func PromptConfirm(prompt string, args ...interface{}) bool {
+	for {
```
[the remark this answers is not in the document]
Comment (oshankkumar) [reply glustercli/cmd/utils.go]: if  user has given some other input instead of yes/no then it will prompt again

File: Gopkg.lock (hunk 5 above)
Comment (oshankkumar) [reply Gopkg.lock]: this is autogenerated

Comment (aravindavk) [reply Gopkg.lock]: You can revert this change, I think it is changed because of external lib used earlier

File: glustercli/cmd/utils.go (hunk 3 above)
Comment (aravindavk) [reply glustercli/cmd/utils.go]: ignore my comment about bufio, `Scanln` also looks good

File: Gopkg.lock (hunk 5 above)
Comment (oshankkumar) [reply Gopkg.lock]: sure

**Comment by oshankkumar** [comment]
retest this please

**Comment by prashanthpai** [comment]
I'll wait for @Madhu-1's approval before merging

**Comment by oshankkumar** [comment]
@Madhu-1 as suggested by @aravindavk , -y flag will become similar to dnf and other commands.
